### PR TITLE
[UX] Fix the output skipped lines

### PR DESCRIPTION
**Pull request by Michaelvll**
Blocked by #3575
In #3572, we changed the skip lines for `setup` command, which accidentally hide some lines in the run output. We now make the skip_lines more robust and add a smoke test for it.

Master:
`sky launch -c test-output --cloud gcp --cpus 2`
```
I 05-23 08:45:47 cloud_vm_ray_backend.py:3212] Running setup on 1 node.
Warning: Permanently added '34.134.84.60' (ECDSA) to the list of known hosts.
running setup
I 05-23 08:45:50 cloud_vm_ray_backend.py:3225] Setup completed.
I 05-23 08:45:57 cloud_vm_ray_backend.py:3314] Job submitted with Job ID: 1
INFO: Waiting for task resources on 1 node. This will block if the cluster is full.
INFO: All task resources reserved.
INFO: Reserved IPs: ['10.128.0.32']
(task, pid=3789) finish
INFO: Job finished (status: SUCCEEDED).
Shared connection to 34.134.84.60 closed.
I 05-23 08:28:19 cloud_vm_ray_backend.py:3354] Job ID: 1
```

This PR:
```
I 05-23 08:28:05 cloud_vm_ray_backend.py:3217] Running setup on 1 node.
hi
I 05-23 08:28:07 cloud_vm_ray_backend.py:3230] Setup completed.
I 05-23 08:28:13 cloud_vm_ray_backend.py:3319] Job submitted with Job ID: 2
I 05-23 08:28:15 log_lib.py:408] Start streaming logs for job 2.
INFO: Tip: use Ctrl-C to exit log streaming (task will not be killed).
INFO: Waiting for task resources on 1 node. This will block if the cluster is full.
INFO: All task resources reserved.
INFO: Reserved IPs: ['10.128.0.30']
(task, pid=6090) hi
INFO: Job finished (status: SUCCEEDED).
I 05-23 08:28:19 cloud_vm_ray_backend.py:3354] Job ID: 2
```

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [x] All smoke tests: `pytest tests/test_smoke.py --kubernetes` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
  - [x] `pytest tests/test_smoke.py::test_minimal`
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: sky/backends/cloud_vm_ray_backend.py [hunk 1]
```diff
@@ -3177,6 +3177,11 @@ def _setup_node(node_id: int) -> None:
                 process_stream=False,
                 # We do not source bashrc for setup, since bashrc is sourced
                 # in the script already.
+                # Skip three lines due to the /bin/bash -i and source ~/.bashrc
```
Comment (cblmemo) [inline sky/backends/cloud_vm_ray_backend.py:3180]:
```suggestion
                # Skip two lines due to the /bin/bash -i and source ~/.bashrc
```

File: sky/utils/command_runner.py [hunk 2]
```diff
@@ -228,6 +234,12 @@ def run(
             ssh_mode: The mode to use for ssh.
                 See SSHMode for more details.
             separate_stderr: Whether to separate stderr from stdout.
+            connect_timeout: timeout in seconds for the ssh connection.
+            source_bashrc: Whether to source the bashrc before running the
```
Comment (cblmemo) [inline sky/utils/command_runner.py:238]:
```suggestion
            source_bashrc: Whether to source the ~/.bashrc before running the
```

File: sky/utils/command_runner.py [hunk 3]
```diff
@@ -228,6 +234,12 @@ def run(
...
+                command.
+            skip_lines: The number of lines to skip from the output. This is
```
Comment (cblmemo) [inline sky/utils/command_runner.py:240]: explicitly saying that we are skipping the **first** `skip_lines` lines?

File: tests/test_smoke.py [hunk 4]
```diff
@@ -270,34 +270,75 @@ def test_example_app():
...
+    'echo "==Validating task output ending 2==" && '
+    'echo "$s" | grep -A 1 "INFO: Job finished (status: SUCCEEDED)" | '
+    'grep "Job ID:"')
+
+
 # ---------- A minimal task ----------
 def test_minimal(generic_cloud: str):
     name = _get_cluster_name()
+    validate_output = _VALIDATE_LAUNCH_OUTPUT
+    # Kubernetes will have
```
Comment (cblmemo) [inline tests/test_smoke.py:309]: will have?

**Review by cblmemo — APPROVED** [review]
Thanks for identifying and fixing this @Michaelvll ! It looks great and should be ready to go after the format test passed ;)

File: sky/utils/command_runner.py (hunk 3 above)
Comment (Michaelvll) [reply sky/utils/command_runner.py]: Added. Thanks!

**Pull request merged**